examples/scripts/project/categorization/bootstrap_taxonomy.py

```diff
@@ -0,0 +1,169 @@
+LOGGER = logging.getLogger(__name__)
+
+CATEGORY_ATTRIBUTE_NAME = "Category Name"
+FULL_PATH_LIST_NAME = "Full Path List"
+FULL_PATH_STR_NAME = "Full Path"
+COLUMN_NAMES = [FULL_PATH_STR_NAME, FULL_PATH_LIST_NAME]
+PK_NAME = "Primary Key"
+
+
+def main(*, instance_connection_info: Dict[str, Any], categorization_project_id: str) -> bool:
+    """
+    Bootstraps taxonomy in a categorization project
+
+    Args:
+        instance_connection_info: Information for connecting to Tamr (host, port, username etc)
+        categorization_project_id: The id of the target categorization project
+
+    Returns: Boolean indicating whether boostrap was successful or not
+
+    Raises:
+        TypeError: retrieved project is not a categorization project
+
+    """
+
+    # Create the tamr client
+    tamr_client = tbox.utils.client.create(**instance_connection_info)
+
+    # Retrieve the project
+    project = tamr_client.projects.by_resource_id(categorization_project_id)
+    LOGGER.info(f"Retrieved project with name: {project.name}")
+
+    # Get the project taxonomy
+    # NOTE: .taxonomy() function breaks on newer tamr instances SUP-5075
+    try:
+        project.taxonomy()
+    except requests.exceptions.RequestException:
+        no_taxonomy_error = f"Project {project.name} is not associated with any taxonomy yet."
+        raise RuntimeError(no_taxonomy_error)
+    LOGGER.info(f"Retrieved project taxonomy with name: {project.taxonomy().name}")
+
+    # Bootstrap all available categories
+    categories = project.taxonomy().categories()
+    category_list = []
+    for category in categories:
+        category_list.append(category.path)
+    # For bootstrapping at a specific tier = -1 (leaf), 1, 2, ... use the existing function
+    # category_set = metrics._get_categories_at_tier(project=project, tier=1)
+    # category_list = [category.split("|") for category in category_set]
+    category_list.sort()
+
+    # Create a dataset with taxonomy categories
+    dataset_name = f"{project.unified_dataset().name}_taxonomy_bootstrap_dataset"
+    try:
+        project.client.datasets.by_name(dataset_name)
+    except KeyError:
+        # Dataset with `dataset_name` does not exist in Tamr. Proceed with dataset creation.
+        pass
+    else:
+        dataset_exists_error = (
+            f"A dataset with name {dataset_name} already exists. Try again after deleting the "
+            "dataset."
+        )
+        LOGGER.error(dataset_exists_error)
+        raise RuntimeError(dataset_exists_error)
+
+    # Create a dictionary of full path between string and list (used as label path)
+    taxonomy_dict = {", ".join(category): category for category in category_list}
+
+    # Create a dataframe
+    df = pd.DataFrame(list(taxonomy_dict.items()), columns=COLUMN_NAMES)
+
+    # Add category data and hash for tamr_id
+    df[CATEGORY_ATTRIBUTE_NAME] = df[FULL_PATH_LIST_NAME].apply(lambda x: x[-1])
+    df[PK_NAME] = df[FULL_PATH_STR_NAME].apply(lambda x: hash(x))
```

Comment:
`hash` is not guaranteed to be the same on multiple runs. If you creating a hash you should use `hashlib` instead. However, ideally you should not use any hashing at all. Instead, the primary key can simply be the FULL_PATH_STR_NAME. Then let Tamr create the `tamr_id` and retrieve that directly. See also my comment below

Reply:
Agreed that the `hash` is not necessary here, however, letting Tamr create the `tamr_id` and retrieving it is not scale-able, I think.  The only way to get the `tamr_id` of a record from Tamr using versioned APIs is to stream the entire dataset, which won't always be reasonable for very large projects.  Until there is a way to query for specific records using versioned APIs (here looking for the `origin_source_name` value), using a transformation in this way to set the `tamr_id` might be the only good solution.